src/main/java/org/spongepowered/common/data/utils/items/SpongeAuthorProcessor.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.utils.items;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.spongepowered.common.data.DataTransactionBuilder.fail;
+import static org.spongepowered.common.data.DataTransactionBuilder.successNoData;
+
+import com.google.common.base.Optional;
+import net.minecraft.init.Items;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagString;
+import org.spongepowered.api.data.DataHolder;
+import org.spongepowered.api.data.DataPriority;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.DataView;
+import org.spongepowered.api.data.manipulator.item.AuthorData;
+import org.spongepowered.api.service.persistence.InvalidDataException;
+import org.spongepowered.api.text.Texts;
+import org.spongepowered.common.data.SpongeDataProcessor;
+import org.spongepowered.common.data.manipulators.items.SpongeAuthorData;
+
+public class SpongeAuthorProcessor implements SpongeDataProcessor<AuthorData> {
+
+    @Override
+    public Optional<AuthorData> getFrom(DataHolder holder) {
+        if (!(holder instanceof ItemStack)) {
+            return Optional.absent();
+        }
+        if (((ItemStack) holder).getItem() != Items.written_book) {
+            return Optional.absent();
+        }
+        if (!((ItemStack) holder).hasTagCompound()) {
+            return Optional.absent();
+        }
+        if (!((ItemStack) holder).getTagCompound().hasKey("author")) {
+            return Optional.absent();
+        }
+        final String rawAuthor = ((ItemStack) holder).getTagCompound().getString("author");
+        return Optional.of(create().setValue(Texts.of(rawAuthor)));
```

Comment:
I do believe this should be `Texts.fromLegacy(rawAuthor)` since it is possible to store the author as a formatted string object.